lib/server.js

```diff
@@ -208,6 +209,29 @@ function listenPromise (server, listenOptions) {
   })
 }
 
+// This function used to compile a validation function for http version
+// bypass here is used to bypass the version check
+function compileValidateHTTPVersion (options) {
+  let bypass = false
+  const obj = {}
+  if (options.serverFactory) {
+    bypass = true
+  }
+  if (options.http2) {
+    obj['2.0'] = true
+    if (options.https && options.https.allowHTTP1 === true) {
+      obj['1.1'] = true
+      obj['1.0'] = true
+    }
+  } else {
+    obj['1.1'] = true
+    obj['1.0'] = true
+  }
+  return function validateHTTPVersion (httpVersion) {
+    return bypass || httpVersion in obj
```

Comment:
`httpVersion in obj` here provide same performance as `httpVersion === '2.0' || httpVersion === '1.1' || httpVersion === '1.0'`